src/main/java/org/apache/sling/scripting/core/impl/jsr223/SlingScriptEngineManager.java

```diff
@@ -237,17 +273,79 @@ public void bundleChanged(BundleEvent event) {
     }
 
     @Activate
-    private void activate(final BundleContext bundleContext) {
+    private void activate(final Config config, final BundleContext bundleContext) {
+        String[] includes = config.includes();
+        if (includes == null) {
+            this.includePatterns = Collections.emptySet();
+        } else {
+            this.includePatterns = new HashSet<>();
+            for (String pattern : includes) {
+                if (!pattern.isEmpty()) {
+                    Pattern p = Pattern.compile(pattern);
+                    includePatterns.add(p);
+                }
+            }
+        }
+
+        String[] excludes = config.excludes();
+        if (excludes == null) {
+            this.excludePatterns = Collections.emptySet();
+        } else {
+            this.excludePatterns = new HashSet<>();
+            for (String pattern : excludes) {
+                if (!pattern.isEmpty()) {
+                    Pattern p = Pattern.compile(pattern);
+                    excludePatterns.add(p);
+                }
+            }
+        }
+
         this.bundleContext = bundleContext;
         bundleContext.addBundleListener(this);
         updateFactories();
     }
 
+    /**
+     * Check if the given factory matches any of the include/excludes patterns
+     *
+     * @param sef the factory to check
+     * @return true if included, false otherwise
+     */
+    private boolean isIncluded(ScriptEngineFactory sef) {
```

Comment:
I think we need some DEBUG logs in this method, to be able to understand why a script engine is included or not

Reply:
Added some debug statements.  Advise if you were looking for something more.

Reply:
Maybe log which pattern matches? Sorry to be picky... Many hours lost to regexp issues over time :-)

Reply:
I added the pattern to the log messages.  But, I can't imagine that there would be much mystery about the matches in this use case as the regex patterns should be quite simple.